sim/dex-formats.ts

```diff
@@ -596,11 +611,23 @@ export class DexFormats {
 				ruleTable.minSourceGen = subRuleTable.minSourceGen;
 			}
 		}
+		this.getTagRules(ruleTable);
 
 		if (!repeals) format.ruleTable = ruleTable;
 		return ruleTable;
 	}
 
+	getTagRules(ruleTable: RuleTable) {
+		const tagRules = [];
+		for (const ruleid of ruleTable.keys()) {
+			if (ruleid.startsWith('+pokemontag:') || ruleid.startsWith('-pokemontag:') || ruleid.startsWith('*pokemontag:')) {
+				if (ruleid.slice(12) === 'allpokemon') continue;
+				tagRules.push(ruleid);
+			}
+		}
+		ruleTable.tagRules = tagRules;
```

Comment:
Note to self: This should be sorted so obtainability rules are always highest priority.

`-Unobtainable, +Legendary` should not let you use Arceus in Gen 1, for instance – `-Unobtainable` must trump every rule that isn't an obtainability rule like `+Past`.